core/src/main/java/org/apache/carbondata/core/util/SegmentMinMaxStats.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.carbondata.core.util;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Holds list of block level min max for each segment
+ */
+public class SegmentMinMaxStats {
+
+  private SegmentMinMaxStats() {
+  }
+
+  public static SegmentMinMaxStats getInstance() {
+    return segmentMinMaxStats;
+  }
+
+  private Map<String, List<SegmentMinMax>> segmentMinMaxMap = new HashMap<>();
+
+  private static final SegmentMinMaxStats segmentMinMaxStats = new SegmentMinMaxStats();
+
+  public Map<String, List<SegmentMinMax>> getSegmentMinMaxMap() {
+    return segmentMinMaxMap;
+  }
+
+  public void setSegmentMinMaxList(String segmentId, Map<String, byte[]> minValues,
+      Map<String, byte[]> maxValues) {
+    if (this.segmentMinMaxMap.get(segmentId) == null) {
+      List<SegmentMinMax> segmentMinMaxList = new ArrayList<>();
+      segmentMinMaxList.add(new SegmentMinMax(minValues, maxValues));
+      this.segmentMinMaxMap.put(segmentId, segmentMinMaxList);
+    } else {
+      this.segmentMinMaxMap.get(segmentId)
```

Comment:
in if check already `this.segmentMinMaxMap.get(segmentId)` is null, so here put the segment id in map, nullpointer can come